### PR TITLE
Parse the timeout parameter as double in the blocking command 

### LINKED ISSUE #1503: Convert the timeout parameter from an integer to a double value
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.

### Motivation

In redis, the timeout parameter of commands like [BLPOP](https://redis.io/commands/blpop/) is a double value, but in kvrocks, the parameter must be an integer.

### Solution

We can change `parseint` in the Parse function into `parsefloat`.

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by Yangsx-1**
Close #1503 
Convert timeout parameter in commands blpop/brpop, bzpopmin/max and bzmpop from int to double.

### CODE REVIEW

File: src/commands/cmd_list.cc [hunk 1]
```diff
@@ -284,7 +286,7 @@ class CommandBPop : public Commander,
 
  private:
   bool left_ = false;
-  int timeout_ = 0;  // seconds
+  int64_t timeout_ = 0;  // microsecond
```
Comment (torwig) [inline src/commands/cmd_list.cc]: `microsecond` => `microseconds` with `s` and in similar places.

File: src/commands/cmd_list.cc [hunk 2]
```diff
@@ -165,16 +165,16 @@ class CommandBPop : public Commander,
...
-      return {Status::RedisParseErr, "timeout is not an integer or out of range"};
+      return {Status::RedisParseErr, errTimeoutIsNotFloat};
     }
 
     if (*parse_result < 0) {
       return {Status::RedisParseErr, "timeout should not be negative"};
     }
 
-    timeout_ = *parse_result;
+    timeout_ = static_cast<int64_t>(*parse_result * 1000 * 1000);  // microsecond
```
Comment (torwig) [inline src/commands/cmd_list.cc]: I think here you can remove the comment. In same places as well.

**Comment by Yangsx-1** [comment]
@torwig There is something wrong in CI with Darwin Clang, what can i do to deal with it?

**Comment by torwig** [comment]
@Yangsx-1 You can try casting `timeout_microsecond` to `int` by using static_cast<int>(timeout_microsecond) because according to the warning, the second argument should be `int` not `int64`.

**Comment by Yangsx-1** [comment]
> @Yangsx-1 You can try casting `timeout_microsecond` to `int` by using static_cast(timeout_microsecond) because according to the warning, the second argument should be `int` not `int64`.

The range of int is a little small, i think using millisecond is better than microsecond.

**Comment by torwig** [comment]
@Yangsx-1 According to this:

```
struct timeval {
   time_t      tv_sec;   // Number of whole seconds of elapsed time
   long int    tv_usec;  // Number of microseconds of rest of elapsed time minus tv_sec. Always less than one million
};
```

The number of microseconds is always less than one million so `int` is OK. Or what is your concern?

**Comment by Yangsx-1** [comment]
> @Yangsx-1 You can try casting `timeout_microsecond` to `int` by using static_cast(timeout_microsecond) because according to the warning, the second argument should be `int` not `int64`.

The range of int is a little small, there may be a chance to overflow for microsecond, but it seems like we can only use int.

**Comment by Yangsx-1** [comment]
> @Yangsx-1 According to this:
> 
> ```
> struct timeval {
>    time_t      tv_sec;   // Number of whole seconds of elapsed time
>    long int    tv_usec;  // Number of microseconds of rest of elapsed time minus tv_sec. Always less than one million
> };
> ```
> 
> The number of microseconds is always less than one million so `int` is OK. Or what is your concern?

Oh, i understand. There is no need to worry about overflow!

**Review by torwig — DISMISSED** [review]
LGTM.
@Yangsx-1 Thank you for your contribution.

**Review by git-hulk — DISMISSED** [review]

File: src/commands/cmd_list.cc [hunk 3]
```diff
@@ -203,7 +203,9 @@ class CommandBPop : public Commander,
 
     if (timeout_) {
       timer_.reset(NewTimer(bufferevent_get_base(bev)));
-      timeval tm = {timeout_, 0};
+      int64_t timeout_second = timeout_ / 1000 / 1000;
+      int64_t timeout_microsecond = timeout_ - timeout_second * 1000 * 1000;
```
Comment (PragmaTwice) [inline src/commands/cmd_list.cc]:
```suggestion
      int64_t timeout_microsecond = timeout_second % (1000 * 1000);
```

**Review by torwig — APPROVED** [review]

**Review by git-hulk — APPROVED** [review]

**Comment by git-hulk** [comment]
Thanks all, merging...

**Pull request merged**